src/Symfony/Component/DependencyInjection/Compiler/AutowirePass.php

```diff
@@ -292,11 +292,11 @@ private function autowireMethod(\ReflectionFunctionAbstract $reflectionMethod, a
             }
 
             if ($checkAttributes) {
-                foreach ($parameter->getAttributes() as $attribute) {
-                    if (\in_array($attribute->getName(), [TaggedIterator::class, TaggedLocator::class, Autowire::class, MapDecorated::class], true)) {
+                foreach ([TaggedIterator::class, TaggedLocator::class, Autowire::class, MapDecorated::class] as $attributeClass) {
+                    foreach ($parameter->getAttributes($attributeClass, \ReflectionAttribute::IS_INSTANCEOF) as $attribute) {
```

Comment:
`Autowire::class === $attributeClass ? \ReflectionAttribute::IS_INSTANCEOF : 0` so that the other attributes are not affected